api/preprints/serializers.py

```diff
@@ -143,5 +149,41 @@ class PreprintDetailSerializer(PreprintSerializer):
     id = IDField(source='_id', required=True)
     subjects = JSONAPIListField(required=False, source='preprint_subjects')
 
+
 class PreprintDetailRetrieveSerializer(PreprintDetailSerializer):
     subjects = JSONAPIListField(required=False, source='get_preprint_subjects')
+
+
+class PreprintPreprintProviderRelationshipSerializer(ser.Serializer):
+    id = ser.CharField(source='_id', required=False, allow_null=True)
+    type = TypeField(required=False, allow_null=True)
```

Comment:
This isn't JSONAPI compliant; they should both be required and not `allow_null` 

>A resource object **MUST** contain at least the following top-level members:
>
> * id
> * type

\- http://jsonapi.org/format/#document-resource-objects